sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -757,7 +758,7 @@ Azure::Response<Models::SubmitTransactionResult> TableClient::SubmitTransaction(
     Core::Context const& context)
 {
   auto url = m_url;
-  url.AppendPath("$batch");
+  url.AppendPath(Azure::Core::Url::Encode("$batch"));
```

Comment:
If we encode `$` as part of the path, do the semantics remain the same? If so, the `$batch` string should be pre-encoded and saved as a constant to avoid multiple encoding attempts.